app/src/main/java/edu/unikom/herbamedjabar/viewModel/AuthViewModel.kt

```
@@ -5,110 +5,151 @@ import androidx.lifecycle.LiveData
+            }
+            _authState.value = AuthState.Loading
+            try {
+                block()
+                Log.d(TAG, "$opName: Authenticated")
+                _authState.value = AuthState.Authenticated
+            } catch (e: kotlinx.coroutines.TimeoutCancellationException) {
+                Log.w(TAG, "$opName: Timeout", e)
+                _authState.value = AuthState.Error(e.toUserMessage("$opName gagal"))
+            } catch (e: kotlinx.coroutines.CancellationException) {
+                throw e
+            } catch (t: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)